src/openrct2/paint/track/Segment.cpp

```diff
@@ -0,0 +1,23 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include "Segment.h"
+
+namespace BlockedSegments
+{
+    const uint16_t kStraightFlat = EnumsToFlags(PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::bottomLeftSide);
+    const uint16_t kDiagStraightFlat[] = {
+        EnumsToFlags(
+            PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::bottomRightSide, PaintSegment::rightCorner),
+        EnumsToFlags(PaintSegment::centre, PaintSegment::topRightSide, PaintSegment::topLeftSide, PaintSegment::topCorner),
+        EnumsToFlags(
+            PaintSegment::bottomLeftSide, PaintSegment::centre, PaintSegment::bottomCorner, PaintSegment::bottomRightSide),
+        EnumsToFlags(PaintSegment::bottomLeftSide, PaintSegment::centre, PaintSegment::leftCorner, PaintSegment::topLeftSide),
+    };
+} // namespace BlockedSegments
```

Comment:
I think its best to have this in the header and declare it constexpr, this is not a lot of data and I don't think having getter functions for those makes much sense.